### ISSUE TITLE
Need help getting Trident work for a functional test for DHS

### ISSUE THREAD

**Issue by rgrmule** [tridentli/trident#144]
Hi team,

I am part of a non-profit that has been engaged by DHS to evaluate their investment in the trident portal and see what is needed to get it more widely used. 

But, I have been unsuccessful in getting it to work. So, I brought in someone who knows Linux much better than I (not a high bar, mind you) - and he struck out as well.

I'm sure it's some documentation thing, but am frankly stymied. 

Failure notes:
I have tried to load the master, as well as 1.4.6 r2 https://github.com/tridentli/trident/releases/tag/DEV_1.4.6-RC2 

I know this is something easy, and I welcome your mirth...
-t

**Comment by teward** [comment]
I would be happy to provide any assistance I can.  I regularly have to deploy both a production instance and test instances for my workplace, and I would be happy to assist where I can in getting you up and running.

Without specific details about what errors you are hitting and what environment you tested in, though, anything I provide will start out as a "from scratch" set up process.

**Comment by rgrmule** [comment]
fantastic! I am starting from scratch and wanted to make it function in a virtual environment before I put it on our server.

when/how do you want to teach this old dog a new trick... or 20?

**Comment by teward** [comment]
@rgrmule My apologies for the slower response time, I was on my commute home and once home I discovered my network decided to cease functioning, which required my time to repair.  I also apologize for my fairly lengthy reply that follows.

Before I go and give you an answer, let me give you a brief 'warning' - to make Trident function in a virtual environment that doesn't have a public IP available to it will have some... minor headaches.  The Trident platform has multiple components, which include email integration for password delivery, nominating new users, and other functions.  My test environments utilize a handful of semi-dedicated public IP addresses with I work with to run live tests (with the email components of the Trident platform) working.  To work in a pure virtual 'lab' deployment, as I call private non-Internet-facing test environments, you'll need to do a number of other environment-configuring bits, including setting up a 'private' network, 'private' DNS server/records for a private test domain, a private mail server to receive those messages, and other components as well; the list of potential modifications to make this 'function' properly can get quite lengthy.

Now, as to the response to your inquiry.  After an off-issue discussion with @bapril via email off-issue regarding certain components and the code sharing restrictions, I can only give you a certain percentage of my workflow to make a 'from scratch' deployment freely and without restriction.  Any such walk-through and workflow I would provide for "how to deploy Trident" will only be able to touch upon the information already in the public view, but with a bit more verbosity and depth, which would get you most of the way to a completely functional Trident instance.  The aforementioned workflow I have already is currently in 'code' form and adapted for the environments I work with, so I will have to take that and make it human-readable; that walk-through will go through all the requisite software, steps, commands, and modifications necessary to get a portal to a 'mostly functioning' state.

I would be happy to discuss a little bit more of the specifics of what will be missing from the workflow in more detail off-issue, and for that please feel free to email me at the address displayed on my GitHub profile page.  From there, we can discuss some of those specific details as to what will be missing from the walkthrough and other related issues that crop up.

------

@bapril if you wish to go into further details than I've done here regarding the bits which you and I exchanged emails on earlier, feel free to do so.

**Comment by rgrmule** [comment]
will do and I'm not wedded on a virtual-only testing environment. Thanks for the detailed response... helps me understand what I'm facing.

**Comment by rgrmule** [comment]
reached out. bounced. 

reached out again... bounced. 

reached out directly (I think).

If you've seen nothing, my email is tmuehleisen @ globalbr.org

My linked in is /tmuehleisen (if you want to confirm I'm really doing this for a non-profit)

**Comment by teward** [comment]
Yep, I've got your email, I'm in a couple of meetings at the moment so my response will be delayed.

**Comment by rgrmule** [comment]
cool! wasn't sure that would work. thanks for the confirmation. any time you have would be appreciated.
